storage/innobase/include/fsp0fsp.ic

```diff
@@ -321,6 +321,7 @@ inline bool fsp_flags_is_valid(uint32_t flags) {
   bool has_data_dir = FSP_FLAGS_HAS_DATA_DIR(flags);
   bool is_shared = FSP_FLAGS_GET_SHARED(flags);
   bool is_temp = FSP_FLAGS_GET_TEMPORARY(flags);
+  bool is_encryption = FSP_FLAGS_GET_ENCRYPTION(flags);
```

Comment:
before this change, how did this function compile without creating the variable `is_encryption`?

Reply:
This has required as https://github.com/mysql/mysql-server/blob/mysql-8.0.29/storage/innobase/include/fsp0fsp.ic#L376 brought over:
```
  /* Only single-table and general tablespaces and not temp tablespaces
  use the encryption clause. */
  if (is_encryption && (is_temp)) {
    return (false);
  }
```
To be on the safe-side, since we do support encryption of temp table, I've kept that if and the is_encryption flag check under `ifndef XTRABACKUP`